_posts/2020-05-03-static-ipv6-playground.md

```diff
@@ -0,0 +1,121 @@
+---
+title: "Static IPv6 Playground"
+date: 2020-05-03
+categories:
+  - blog
+tags:
+  - ipv6
+---
+
+This post is all about setting up an IPv6 enviorment for learning and testing 
+of varios setups and scenarios.
+
+We will use network namespaces for our setup and proceed to define static
+IP addresses.
+(we will cover dynamic IP addresses in a following post)
+
+## Basic Setup
+In order to prepare the ground for our IPv6 playground, we will use network
+namespaces to simulate different network stacks.
+A network namespace is an isolated network stack that includes interfaces,
+ip addresses and routes.
+
+Throught this post, we will use the `ip` command which is part of the
+[iproute2](https://wiki.linuxfoundation.org/networking/iproute2) utilities.
+
+### Namespace creation
+For our setup, we will use two namespaces: red & blue.
+
+Lets create our two namespaces:
+```
+sudo ip netns add red
+sudo ip netns add blue
+```
+Each namespace is created with a loopback interface which requires an explicit
+enablement:
+```
+sudo ip netns exec red ip link set lo up
+sudo ip netns exec blue ip link set lo up
+```
+
+### L2 connectivity (the veth)
+In order to enable connectivity between the two namespaces, we will use a
+veth interface. A veth interface type comes always in pairs, anything that
+ingress one edge, egress the other edge and vice versa. It provides a L2 local
+connectivity between the peers.
+
+We will create the veth interface at the root namespace and then place each
+peer in one of the namespaces.
+
+Lets create the veth interface:
+```
+sudo ip link add veth00 type veth peer name veth10
+```
+And place each in the relevant namespace:
+```
+sudo ip link set dev veth00 netns red
+sudo ip link set dev veth10 netns blue
+```
+
+At this point, assuming that IPv6 is enabled on the host, each namespace
+should show an loopback interface and another veth type interface which
+has a [link-local IPv6 address](https://tools.ietf.org/html/rfc4291#page-11).
+
+To check the interfaces and their addresses, execute:
+```
+sudo ip netns exec red ip addr
```

Comment:
maybe worth to show the output of the commands ?

Reply:
Too much noise IMO and I see no benefit for it. Running it is simple enough if one wants to see it in action.

Reply:
I agree with Or here. I don't think it's noise, I think it's useful information so the user knows what valid output to expect, and can recognize if the command ends with an invalid output (running the command might result in non-error output, but still not the expected one).

Reply:
I agree with Or and Yossi. It worth having the output here and using the output for an example for line 89.

Reply:
The customer is always right. Done.